### ISSUE TITLE
Sync wrong when copy/pasting event in Outlook

### ISSUE THREAD

**Issue by m-rm** [phw198/OutlookGoogleCalendarSync#487]
**OGCS Version**: v2.7.0.0 Beta
**Installed or Portable**: Installed

**Problem Description**  
It seems that when an event is copied in Outlook and then pasted to another date, the date and time of the original event is used when syncing, not of the new event. This means on the synced calender I now have 2 events at the same date and time, but none at the new data and time.

I'm not sure if this is a limitation of the Outlook API or a bug.

**Steps to Reproduce the Issue**  
1. Create an event on any date and time.
2. Sync
3. Click on the event and press Ctrl+C.
4. Click on any other day in the calendar and press Ctrl+V. (change the time if needed, doesn't effect the outcome)
5. Sync
6. In the Google calender there is now a duplicate event, but no new event

**Any other information**  
Using Outlook 2016 32-bit

Log file:
[OGcalsync_#487.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/1810929/OGcalsync_.487.log)

**Comment by phw198** [comment]
I would need to see your log file - after the paste an event should fire to clean the pasted copy of any metadata related to having already been synced.

Also, you don't say, but I'm assuming you are two way syncing?

**Comment by m-rm** [comment]
I attached the log to the description. I created a "Test" event on 21.03.2018 and copied it to 22.03.2018.

I also found out that syncing between creating the original event and the copied event is neccessary to reproduce this behavior. If I don't sync, the Google calendar entries are correct!

No, I'm using one way sync Outlook -> Google.

**Comment by phw198** [comment]
OK, so I think your problem lies here:
```
018-03-14 12:54:40,223 INFO  16 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  Item paste event caught.
2018-03-14 12:54:40,226 DEBUG  16 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  We've got 1 items selected for copy.
2018-03-14 12:54:40,234 DEBUG  16 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  21.03.2018 08:00 => "Test"
2018-03-14 12:54:40,236 DEBUG  16 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  This item isn't managed by OGCS.
2018-03-14 12:54:42,245 DEBUG  17 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  Tagging copied Outlook items
2018-03-14 12:54:42,245 DEBUG  17 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  Finding 1 recently copied and pasted item(s) since 14.03.2018 12:54:40.223...
2018-03-14 12:54:42,258 WARN  17 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  Could not find any Outlook items for post-processing!
```
Could you bump the logging level up to `FINE` and then repeat your test. After the copy event, you should get a line like:
`Filter string: [End] >= '01/01/0001 00:00' AND [Start] < '31/12/9999 23:59' AND [Modified] >= '18/03/2018 14:32'`
Take the modified timestamp and plug this into an Outlook search and test it:
![image](https://user-images.githubusercontent.com/11843015/37565934-d591699c-2aba-11e8-9520-3d091aeb8f24.png)

Can you play with the timestamp until it returns the item you pasted - let me know what the value is.

**Comment by m-rm** [comment]
```
2018-03-27 15:26:14,652 INFO  16 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  Item paste event caught.
2018-03-27 15:26:14,656 DEBUG  16 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  We've got 1 items selected for copy.
2018-03-27 15:26:14,663 FINE  16 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  GetSummary - not all day event
2018-03-27 15:26:14,663 DEBUG  16 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  03.04.2018 08:00 => "Test"
2018-03-27 15:26:14,666 DEBUG  16 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  This item isn't managed by OGCS.
2018-03-27 15:26:16,674 DEBUG  11 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  Tagging copied Outlook items
2018-03-27 15:26:16,674 DEBUG  11 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  Finding 1 recently copied and pasted item(s) since 27.03.2018 03:26:14.653...
2018-03-27 15:26:16,679 FINE  11 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  292 calendar items exist.
2018-03-27 15:26:16,679 FINE  11 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  Filter string: [End] >= '01.01.0001 00:00' AND [Start] < '31.12.9999 23:59' AND [Modified] >= '27.03.2018 15:26'
2018-03-27 15:26:16,687 FINE  11 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  Filtered down to 0
2018-03-27 15:26:16,687 WARN  11 OutlookGoogleCalendarSync.OutlookOgcs.ExplorerWatcher [0] -  Could not find any Outlook items for post-processing!
```

Searching for 27.03.2018 15:26 doesn't find any results. 27.03.2018 15:17 shows 2 results: The original one and the copied one. It seems that the modification date of the copied appointment is the creation date of the original appointment.

**Comment by phw198** [comment]
:package: This should now be resolved with [hotfix v2.7.2.2.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1863703/v2.7.2.2.zip) ([see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)). 

:arrow_double_up: Please make sure you upgrade to v2.7.2 first before applying the hotfix. Due to a [GitHub change](https://github.com/phw198/OutlookGoogleCalendarSync/issues/497) you will need to [download the latest version directly from the project website](https://phw198.github.io/OutlookGoogleCalendarSync/).

Let me know how it goes!

**Comment by m-rm** [comment]
I can happily report that it works now!

Thank you very much!
